cmd/mcp/k8s/actions.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2025 Stefan Prodan.
+// SPDX-License-Identifier: AGPL-3.0
+
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	fluxcdv1 "github.com/controlplaneio-fluxcd/flux-operator/api/v1"
+)
+
+// Annotate sets annotations on a Kubernetes resource identified by GroupVersionKind, name, and namespace.
+func (k *Client) Annotate(ctx context.Context, gvk schema.GroupVersionKind, name, namespace string, keys []string, val string) error {
+	resource := &metav1.PartialObjectMetadata{}
+	resource.SetGroupVersionKind(gvk)
+
+	objectKey := ctrlclient.ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}
+
+	if err := k.Client.Get(ctx, objectKey, resource); err != nil {
+		return fmt.Errorf("unable to read %s/%s/%s error: %w", gvk.Kind, namespace, name, err)
+	}
+
+	patch := ctrlclient.MergeFrom(resource.DeepCopy())
+
+	annotations := resource.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	for _, key := range keys {
+		annotations[key] = val
+		resource.SetAnnotations(annotations)
+	}
+
+	if err := k.Client.Patch(ctx, resource, patch); err != nil {
+		return fmt.Errorf("unable to annotate %s/%s/%s error: %w", gvk.Kind, namespace, name, err)
+	}
+
+	return nil
+}
+
+// Delete deletes a Kubernetes resource identified by GroupVersionKind, name, and namespace.
+func (k *Client) Delete(ctx context.Context, gvk schema.GroupVersionKind, name, namespace string) error {
+	resource := &metav1.PartialObjectMetadata{}
+	resource.SetGroupVersionKind(gvk)
+	resource.SetName(name)
+	resource.SetNamespace(namespace)
+
+	if err := k.Client.Delete(ctx, resource); err != nil {
+		return fmt.Errorf("unable to delete %s/%s/%s error: %w", gvk.Kind, namespace, name, err)
+	}
+
+	return nil
+}
+
+// ToggleSuspension toggles the suspension of a Flux resource by updating the spec.suspend field.
+func (k *Client) ToggleSuspension(ctx context.Context, gvk schema.GroupVersionKind, name, namespace string, suspend bool) error {
```

Comment:
I'd call a method "ToggleSomething" only if it inverts the current value: if the current is true, sets to false, and if the current is false, sets to true. This method just sets the field to the given value, regardless of the current value

```suggestion
// SetSuspend sets the spec.suspend field of a Flux resource.
func (k *Client) SetSuspend(ctx context.Context, gvk schema.GroupVersionKind, name, namespace string, suspend bool) error {
```

Reply:
This method adds the field or removes it, it's a toggle.